src/utils.ts

```diff
@@ -216,3 +216,44 @@ export function arrayEquals<T>(
   if (left.length !== right.length) return false;
   return left.every((element, index) => element === right[index]);
 }
+
+/**
+ * Type that extends any non-function entity--like a string, number, or array--into a itself and a no-input function that returns it.
+ * Used to interact with objects that could either be functions or static values.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type Delayed<T> = [T] extends [(...args: any) => any]
```

Comment:
I think unknown is better than any here? Maybe?

Reply:
you can't use the rest parameter for args with unknown, and once you're using any for that, why not just use the whole thing

Reply:
but I should document the dumb square brackets thing